plugins/by-name/ltex-extra/default.nix

```diff
@@ -43,26 +43,60 @@ lib.nixvim.plugins.mkNeovimPlugin {
     ] "Log level.";
   };
 
-  extraConfig = cfg: {
-    warnings = lib.nixvim.mkWarnings "plugins.ltex-extra" {
-      when = !config.plugins.lsp.enable;
-      message = ''
-        You have enabled `ltex-extra` but not the lsp (`plugins.lsp`).
-        You should set `plugins.lsp.enable = true` to make use of the LTeX_extra plugin's features.
+  extraConfig =
+    cfg:
+    let
+      inherit (config.plugins.lsp.servers) ltex;
+      attachLua = ''
+        require("ltex_extra").setup(${lib.nixvim.toLuaObject cfg.settings})
       '';
-    };
-
-    plugins.lsp = {
-      servers.ltex = {
-        # Enable the ltex language server
-        enable = true;
+    in
+    {
+      warnings = lib.nixvim.mkWarnings "plugins.ltex-extra" [
+        {
+          when = !config.plugins.lsp.enable;
+          message = ''
+            You have enabled `ltex-extra` but not the lsp (`plugins.lsp`).
+            You should set `plugins.lsp.enable = true` to make use of the LTeX_extra plugin's features.
+          '';
+        }
+        {
+          # TODO: Added 2025-03-30; remove after 25.05
+          # Warn if servers.ltex seems to be configured outside of ltex-extra
+          when =
+            !ltex.enable
+            && (
+              ltex.onAttach.function != attachLua
+              || ltex.onAttach.override != false
+              ||
+                [ ] != collect (v: !(isAttrs v || v == null)) (
+                  # Remove options that wouldn't be null in an empty ltex config
+                  removeAttrs ltex [
+                    "extraSettings"
+                    "package"
+                    "onAttach"
+                    "enable"
+                  ]
+                )
+            );
+          message = ''
+            You seem to have configured `plugins.lsp.servers.ltex` for ltex-extra.
+            It now uses `plugins.lsp.servers.ltex_plus` by default,
+            either move the configuration or explicitly enable `ltex` with
+            `plugins.lsp.servers.ltex.enable = true`
+          '';
+        }
+      ];
 
-        onAttach.function = ''
-          require("ltex_extra").setup(${lib.nixvim.toLuaObject cfg.settings})
-        '';
+      plugins.lsp = {
+        servers.ltex.onAttach.function = attachLua;
+        servers.ltex_plus = {
+          # Enable ltex_plus if ltex is not already enabled
+          enable = mkIf (!ltex.enable) true;
+          onAttach.function = attachLua;
```

Comment:
Both these `onAttach` definitions should also be between `mkDefault` and `mkOptionDefault` priority.

Reply:
I think I disagree with this. Having a different priority means a user may unintentionally disable this plugin by overriding this hook that's calling the setup.